app/uk/gov/hmrc/incometaxproperty/connectors/IntegrationFrameworkConnector.scala

```diff
@@ -44,4 +44,34 @@ class IntegrationFrameworkConnector @Inject()(httpClient: HttpClient, appConf: A
   private def callGetBusinessDetails(url: URL)(implicit hc: HeaderCarrier): Future[GetBusinessDetailsResponse] = {
     httpClient.GET[GetBusinessDetailsResponse](url)
   }
+
+  def getAllPeriodicSubmission(taxYear: String,
+                               taxableEntityId: String,
+                               incomeSourceId: String)
+                              (implicit hc: HeaderCarrier): Future[Either[ApiError, PeriodicSubmissionModel]] = {
+
+    val (url, apiVersion) = if (taxYear.take(2).toInt >= 23) {
+      (new URL(s"${appConfig.ifBaseUrl}/income-tax/business/property/$taxYear/$taxableEntityId/$incomeSourceId/period"), "1954")
+    } else {
+      (new URL(s"$baseUrl/income-tax/business/property/$taxableEntityId/$incomeSourceId/period?taxYear=${toTaxYearParam(taxYear)}"), "1649")
+    }
+
+    val apiResponse = callGetPeriodicSubmissionUpstream(url)(ifHeaderCarrier(url, apiVersion))
+    val dataResponse = apiResponse.map { response =>
+      if (response.result.isLeft) {
+        Left(ApiError(response.httpResponse.status, SingleErrorBody(response.getClass.getSimpleName, response.httpResponse.body)))
+      } else {
+        response.result
+      }
+    }
+    dataResponse
+  }
+
+  private def callGetPeriodicSubmissionUpstream(url: URL)(implicit hc: HeaderCarrier): Future[GetPeriodicSubmissionDataResponse] = {
+    httpClient.GET[GetPeriodicSubmissionDataResponse](url)
+  }
+
+  private def toTaxYearParam(taxYear: String): String = {
+    s"20${taxYear.take(2).toInt - 1}-20${taxYear.takeRight(2)}"
```

Comment:
We can do it the same way we do in state beneftis so we don't need to use the `20`?
https://github.com/hmrc/income-tax-state-benefits/blob/main/app/connectors/IntegrationFrameworkConnector.scala#L184

Reply:
Changed to Int